### ISSUE TITLE
Date "NaN/NaN/NaN"

### ISSUE THREAD

**Issue by dragoncity17** [chriskomus/aliexpress-order-downloader#3]
Hello,
Thanks for answers and update.

**Date**
So for date I have "NaN/NaN/NaN".
Doesn't work for me.
I am french maybe it's that, original it's "Commande passée le: 29 mars 2022".

**Order ID**

For orderid I changed for this for the moment, but maybe there is better.

I changed "Copy" for "Copier" because I am french :)

order.orderId =      orderDateAndId.substring(
                             orderDateAndId.indexOf("ID: ") + 53,
                             orderDateAndId.lastIndexOf("Copier")).trim();

-------------------------------------
Example of extract I use :

        s += "\"" + e.productTitle + "\"\t";
        s += e.productQuantity + "\t";
        s += "\"=HYPERLINK(" + "\"\"" + e.productUrl + "\"\"" + ";" +  "IMAGE(" + "\"\"" + e.productImageUrl + "\"\"" + ";1" + "\"))\t";
        s += e.order.total + "€\t";
        s += e.order.storeName + "\t";
        s += "\"=HYPERLINK(" + "\"\"" + "https://trade.aliexpress.com/order_detail.htm?orderId=" + e.order.orderId + "\"\"" +  ";" + "\"\"" + e.order.orderId + "\"\"" + "\")\t";

**Comment by dragoncity17** [comment]
![alt text](https://snipboard.io/ehqZLK.jpg)

For you understand.

**Comment by chriskomus** [comment]
Thanks! I think I found a fix for the date by using moments.js, which allows using an array of date formats to attempt to create a date object from a string. In this case: ["MMM D, YYYY", "D MMM, YYYY", "MMM DD, YYYY", "DD MMM, YYYY"]
I also updated the code to retrieve the date and order ID without using language specific text to scrape between.

**Comment by dragoncity17** [comment]
Great ! 
I just change for that :
month = order.date.getMonth() + 1
order.dateString =   order.date.getDate() + '/0' + month + '/' + order.date.getFullYear();
        
If not month was not good (always previous month).

And change for total because forget coma "," :
order.total =        $(eo).find(".order-item-content-opt-price-total").text().split('Total: €').pop();

Now it's perfect ;)

**Comment by dragoncity17** [comment]
Hello,
Today I bought something on Ali Express.

And again problem with date, with previous order no problem.

There is again "NaN/NaN/NaN".

Need wait ...?

**Comment by chriskomus** [comment]
Interesting. I wonder if it has something to do with it being the 1st of the month. Do you have this line in the header?
`// @require      https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.29.1/moment.min.js`

And you are using this line for the date?
`order.date =         moment(rawDate, ["MMM D, YYYY", "D MMM, YYYY", "MMM DD, YYYY", "DD MMM, YYYY"]).toDate();`

Do you get a value for order.date or is there an error for that as well as order.dateString?

**Comment by dragoncity17** [comment]
Yes for your 2 questions,in fact I tested with your last update and just use datestring (I prefer).

For my previous order no problem but last order doesn't work.

Maybe because previously I used your Button for load order (you removed this Button).

Try to buy something you will see :-)

**Comment by chriskomus** [comment]
Been a bit busy the last few days so I haven't had a chance to look into it. I'm going to try a few things and maybe try the script in a VM running in French to see if I can replicate that error. I appreciate all the feedback!
Also, I removed the 3rd button because the main copy button essentially does what the 3rd button was doing, so it was redundant.

**Comment by dragoncity17** [comment]
I tested again, all order previous april > WORK.

But since 01/04/2022... all say NaN/0NaN/NaN.

Thanks for your help :)

**Comment by chriskomus** [comment]
I see the problem now. I switched to French (top right corner of the Ali site) and all dates in April and Febuary are broken. March and January work fine. The issue is with the script parsing 'avr.' and 'févr.' and not knowing what to do with it. However, it seems to understand 'mars' and 'janv.'. Not sure why. I will do some more testing.
In the meantime, a workaround would be switching languages on Ali to English.

**Comment by dragoncity17** [comment]
Exactly ! :)
Strange.

If you find solution in french tell me, thanks for your help ! :)
